lightning/src/chain/chaininterface.rs

```diff
@@ -35,15 +35,19 @@ pub trait BroadcasterInterface {
 	fn broadcast_transactions(&self, txs: &[&Transaction]);
 }
 
-/// An enum that represents the speed at which we want a transaction to confirm used for feerate
+/// An enum that represents the priority at which we want a transaction to confirm used for feerate
 /// estimation.
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
 pub enum ConfirmationTarget {
-	/// We are happy with this transaction confirming slowly when feerate drops some.
+	/// We are happy with a transaction eventually confirming, regardless of how long it may take,
```

Comment:
I think to clarify the discussion, it should be precised which mempool min fee `MempoolMinimum` aims to represent. As a recall you have two mempool min fees in Bitcoin Core:
- mempool min fee
- min relay fee 

The first one is dynamic and based on the size of your mempool and the current received transactions (see `GetMinFee` in `txmempool.cpp`).

The second one is static and it can be configured by the node operator (see `m_min_relay_feerate` in `txmempool.cpp`)

Our pre-signed transactions should respect both at all time, at least until bip331 + nversion=3 are widely deployed on the network, which might take one or two Core release cycle.

What is unclear in my understanding is how to evaluate the “maximum mempool-min-fee over the course of the coming year or so” as we’re trying to predict future blockspace demand by all the Bitcoin users. If we hardcode some value and we deploy LDK software, and we got wrong our prediction, in my understanding it would mean all the deployed nodes are vulnerable to loss of funds (as the pre-signed commitment with `MempoolMinimum` won’t satisfy Core’s `GetMinFee` check in `CheckFeeRate`.

Reply:
> Like everything fee-related in lightning, until we have package relay its simply not possible to be "secure". We should communicate that to our users here, and maybe also note that once the user is using package relay,

Yes this is correct it cannot be “secure” until we have package relay. The more intermediate solution would be to scale our `update_fee` frequency on the changes of mempool min fee announced to us by the local mempool, it’s still a “security” advance. Though clearly conversation for another issue / PR.

> Assuming they're not returning some fixed value (which they likely shouldn't yet) is this really necessary? The frequency of update_fee is unchanged, so we should be able to detect any feerate fluctuations.

Sure if they’re using fixed value it’s okay though at commit `7751cb90` `MempoolMinimum` documentation could be clear on that concern, what do you think of the following, I think this is the most accurate we can say until wide deployment of package relay (might take another couple of years)

```
diff --git a/lightning/src/chain/chaininterface.rs b/lightning/src/chain/chaininterface.rs
index 4204d714..b5c05226 100644
--- a/lightning/src/chain/chaininterface.rs
+++ b/lightning/src/chain/chaininterface.rs
@@ -44,8 +44,19 @@ pub enum ConfirmationTarget {
        /// bump of the transaction.
        ///
        /// The feerate returned should be the absolute minimum feerate required to enter most node
-       /// mempools across the network. Note that if you are not able to obtain this feerate estimate,
-       /// you should likely use the furthest-out estimate allowed by your fee estimator.
+       /// mempools across the network. A good implementation can be to poll frequently (e.g every minute)
+       /// Bitcoin Core's `getmempoolinfo` and consume the `mempoolminfee` min fee, as such guaranteeing
+       /// this value reflect roughly real-time network mempools congestion.
+       ///
+       /// In the lack of access to a local mempool, this feerate estimate can be fetched from (ideally)
+       /// a set of third-party hosting local mempools, however this means those third-parties can 
+       /// salvage the propagation of your pre-signed transactions at anytime and therefore endangers
+       /// the safety of channels funds. It SHOULD be considered carefully as a deployment.
+       ///
+       /// A third alternative is to use a furthest-out estimate based on historical worst-case seen
+       /// mempools feerate, and multiply this historical worst in function of the safety margin you
+       /// wish. Failure in the further-out prediction renders the propagation of your pre-signed
+       /// transaction ineffective and therefore endangers the safety of channels funds too.
        MempoolMinimum,
        /// We are happy with a transaction confirming slowly, at least within a day or so worth of
        /// blocks.
```

Reply:
If I remember correctly the spec says nothing about the frequency of the `update_fee`, or at least nothing in a mandatory way, otherwise it would be bad as you have to synchronize clocks, including being based on blocks. Polling does help for the negotiation of yet-to-come state, where yet-to-come state might be defined simply by your counterparty sending `update_add_htlc` / `update_fulfill_htlc` at anytime, without synchronicity on network mempools congestion.

The advice about remote fee estimator starts to be spread multiple times across the feedback, e.g about dust exposure, so yes it could be unified for `FeeEstimator`, and the other hand it might be better to have the critical documentation nearer from where the user are interacting with a feature / variables.

If I had a reliable methodology that one can implement in code about predicting _future_ min-fees and that one can resume in 4 lines of documentation, well I would write them :) To be honest that type of discussions belong more to #2404, it’s a nice thing we can have on top of it.

For this PR, I think an effective documentation might be just to give a hardcoded min-fee based on the last two years of data, with a x2 or x3 (or more ?) safety margin with a suggestion to update it as each release.

Reply:
> Sadly we use this canoe both for our own channels and to validate peers, so we generally can't add a few x multiplier.

Changes nothing as here we wanna to validate the sanity of the mempool min fee, whoever is responsible of the `update_fee`. Of course if your counterparty announces something worst than the multiplier we should reject it, as it means our _future_ min fees prediction are broken. And here we have to consider what we accept as a worst-case frequency from counterparty’s `update_fee` (4 source of logic that can influence on update validity, holder’s mempool/feeestimator, holder’s update_fee frequency, counterparty’s mempool/feestimator, counterparty’s update_fee frequency).